src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -104,5 +178,29 @@ public Task<IQuantumMachineJob> SubmitAsync(IQuantumMachine machine, AzureSubmis
             var submitParameters = new object[] { EntryPointInfo, entryPointInput, submissionContext };
             return (Task<IQuantumMachineJob>)submitMethod.Invoke(machine, submitParameters);
         }
+
+        /// <inheritdoc/>
+        public Task<IQuantumMachineJob> SubmitAsync(IQirSubmitter submitter, AzureSubmissionContext submissionContext, CancellationToken cancellationToken = default)
+        {
+            var entryPointInput = GetEntryPointInputArguments(submissionContext);
+
+            var options = SubmissionOptions.Default;
+            options = options.With(submissionContext.FriendlyName, submissionContext.Shots, submissionContext.InputParams);
+
+            // Find and invoke the method on IQirSubmitter that is declared as:
+            // Task<IQuantumMachineJob> SubmitAsync(Stream qir, string entryPoint, IReadOnlyList<Argument> arguments, SubmissionOptions submissionOptions)
+            var submitMethod = typeof(IQirSubmitter)
+                .GetMethods()
+                .Single(method =>
+                    method.Name == "SubmitAsync"
+                    && method.GetParameters().Length == 4
+                    && method.GetParameters()[0].ParameterType == typeof(Stream)
+                    && method.GetParameters()[1].ParameterType == typeof(string)
+                    && method.GetParameters()[2].ParameterType == typeof(IReadOnlyList<Argument>)
+                    && method.GetParameters()[3].ParameterType == typeof(SubmissionOptions)
+                    );
+            var submitParameters = new object[] { QirStream!, $"ENTRYPOINT__{submissionContext.OperationName}", entryPointInput, options };
```

Comment:
I think the `"ENTRYPOINT"` string literal is used in other places in this code base. It would be good to make it a `static readonly string` somewhere and reference it instead of using a string literal.

Reply:
Knowing where to put that is the tricky part 😄.

Reply:
It is tricky indeed!

Reply:
I've added it as a public static string to CompilerService. However, now we have a public-facing string that is just "ENTRYPOINT". Is that the direction we want to go with this?

Reply:
As per our discussion, I've made the CompilerService's EntrypointNamespaceName member private, and changed the other two locations to use the string literal with a comment that they must be the same as the CompilerService's namespace name.